### PR TITLE
Refactoring reactions to use Redux

### PR DESCRIPTION
**Pull request by deadlybutter**
This PR refactors the reactions components to use Redux

- Adds Reportback Container to connect it with Redux
- Changes up the Reportback Block to reflect the new props structure. 
  - This got quite large, and might be worth breaking up. Thoughts?
- Added some reaction specific actions & reducers to manage that state.
- Updated the reaction component to work with this new structure.

### CODE REVIEW

**Comment by deadlybutter** [comment]
@weerd Ready for a review!

File: resources/assets/actions/index.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ export const STORE_REPORTBACK_PENDING = 'STORE_REPORTBACK_PENDING';
 export const STORE_REPORTBACK_SUCESSFUL = 'STORE_REPORTBACK_SUCESSFUL';
 export const ADD_TO_SUBMISSIONS_LIST = 'ADD_TO_SUBMISSIONS_LIST';
 export const CLICKED_VIEW_MORE = 'CLICKED_VIEW_MORE';
+export const USER_LIKED_REPORTBACK = 'USER_LIKED_REPORTBACK';
+export const USER_UNLIKED_REPORTBACK = 'USER_UNLIKED_REPORTBACK';
+export const REACTION_COMPLETE = 'REACTION_COMPLETE';
```
Comment (weerd) [inline resources/assets/actions/index.js]: Just a thought, could we rename these to `USER_TOGGLED_REACTION_ON` and `USER_TOGGLED_REACTION_OFF`? or something that isn't so specific to the idea of a "like" (the heart)?

File: resources/assets/components/Reaction/index.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ import classnames from 'classnames';
 const Reaction = (props) => {
   const active = props.active || false;
   const total = props.total || 0;
-  const onClick = props.onClick || null;
+
+  const onClick = () => active ? props.onUnlike() : props.onLike();
```
Comment (weerd) [inline resources/assets/components/Reaction/index.js]: Similar to my comment above about naming to be more generic and not so specific to "like". Thoughts?

File: resources/assets/actions/index.js (hunk 1 above)
Comment (deadlybutter) [inline resources/assets/actions/index.js]: 👍  sounds good to me

File: resources/assets/components/Reaction/index.js (hunk 2 above)
Comment (deadlybutter) [inline resources/assets/components/Reaction/index.js]: Yeah Ill carry the `toggle` naming convention down to here as well

File: resources/assets/reducers/reactions.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import {
+  USER_TOGGLED_REACTION_ON,
+  USER_TOGGLED_REACTION_OFF,
+  REACTION_COMPLETE,
+} from '../actions';
+const update = require('react/lib/update');
```
Comment (weerd) [inline resources/assets/reducers/reactions.js]: not using `import`?

Comment (deadlybutter) [inline resources/assets/reducers/reactions.js]: I copied it from a SO answer, but could make it an import to make it fall in line with everything else for sure

Comment (weerd) [inline resources/assets/reducers/reactions.js]: also why are you calling `update()` directly in each return?

Comment (deadlybutter) [inline resources/assets/reducers/reactions.js]: I can stash it in a variable but seemed redundant?

**Comment by deadlybutter** [comment]
@DFurnes Refactored to use async actions!

File: resources/assets/actions/index.js [hunk 4]
```diff
@@ -10,6 +10,10 @@ export const STORE_REPORTBACK_PENDING = 'STORE_REPORTBACK_PENDING';
 export const STORE_REPORTBACK_SUCESSFUL = 'STORE_REPORTBACK_SUCESSFUL';
 export const ADD_TO_SUBMISSIONS_LIST = 'ADD_TO_SUBMISSIONS_LIST';
 export const CLICKED_VIEW_MORE = 'CLICKED_VIEW_MORE';
+export const USER_TOGGLED_REACTION = 'USER_TOGGLED_REACTION';
+export const USER_TOGGLED_REACTION_ON = 'USER_TOGGLED_REACTION_ON';
```
Comment (DFurnes) [inline resources/assets/actions/index.js]: These two constants (`USER_TOGGLED_REACTION_ON` and `USER_TOGGLED_REACTION_OFF`) can be removed, yeah?

Comment (deadlybutter) [inline resources/assets/actions/index.js]: Oh yep, nice catch

**Pull request merged**